### PR TITLE
ROC-3417: make timeline optional for defendant

### PR DESCRIPTION
**Pull request by sabahirfan**
### JIRA link (if applicable) ###
[ROC-3417](https://tools.hmcts.net/jira/browse/ROC-3417)

### Change description ###
This PR makes defendant timeline rows to be optional. Also updates static text on PDF.

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/resources/citizen/templates/document/defendantResponseReceipt.html [hunk 1]
```diff
@@ -338,40 +338,7 @@
         </td>
       </tr>
       {% endif %}
-      {% if defendant.events is defined %}
```
Comment (sabahirfan) [inline src/main/resources/citizen/templates/document/defendantResponseReceipt.html]: Removed this block as it is not required. Luckily the conditions  `{% if defendant.events is defined %` and `{% if defendant.evidences is defined %}` were never true, so we were not seeing duplicate on pdf.

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/DefendantTimeline.java [hunk 2]
```diff
@@ -1,28 +1,39 @@
...
-        super(events);
+        this.events = events;
         this.comment = comment;
     }
 
     public Optional<String> getComment() {
         return Optional.ofNullable(comment);
     }
 
+    public Optional<List<TimelineEvent>> getEvents() {
```
Comment (geek-soft) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/DefendantTimeline.java]: I am not convinced to that, it is good pattern to always have list initialised, if so there is no need to use optional for lists.

File: src/main/resources/citizen/templates/document/defendantResponseReceipt.html [hunk 3]
```diff
@@ -274,17 +274,17 @@
         </td>
       </tr>
 
-      {% if events is defined %}
+      {% if events is defined and events is not empty %}
```
Comment (geek-soft) [inline src/main/resources/citizen/templates/document/defendantResponseReceipt.html]: Isn't `events is not empty` enough? A http://www.mitchellbosecke.com/pebble/documentation/test/empty says that it handle nulls.

File: src/main/resources/citizen/templates/document/defendantResponseReceipt.html [hunk 4]
```diff
@@ -299,17 +299,17 @@
       </tr>
       {% endif %}
 
-      {% if evidences is defined %}
+      {% if evidences is defined and evidences is not empty %}
```
Comment (geek-soft) [inline src/main/resources/citizen/templates/document/defendantResponseReceipt.html]: The same here.

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/response/DefendantTimelineMapper.java [hunk 5]
```diff
@@ -30,6 +32,7 @@ public CCDDefendantTimeline to(DefendantTimeline timeline) {
         CCDDefendantTimeline.CCDDefendantTimelineBuilder builder = CCDDefendantTimeline.builder();
         builder.events(
             timeline.getEvents()
+                .orElse(emptyList())
```
[the remark this answers is not in the document]
Comment (sabahirfan) [reply ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/response/DefendantTimelineMapper.java]: done

File: src/main/resources/citizen/templates/document/defendantResponseReceipt.html (hunk 3 above)
Comment (sabahirfan) [reply src/main/resources/citizen/templates/document/defendantResponseReceipt.html]: done

File: src/main/resources/citizen/templates/document/defendantResponseReceipt.html (hunk 4 above)
Comment (sabahirfan) [reply src/main/resources/citizen/templates/document/defendantResponseReceipt.html]: done

**Comment by geek-soft** [comment]
@dharmendrak please re-review and approve.

**Comment by sabahirfan** [comment]
Demo done to @pragnesh-patel

**Pull request merged**